docs/content/operations/pull-deps.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: doc_page
+---
+# pull-deps Tool
+
+`pull-deps` is a tool that can pull down dependencies to the local repository and lay dependencies out into the extension directory as needed.
+
+`pull-deps` has several command line options, they are as follows:
+
+`-c` or `--coordinate`     
+
+    Extension coordinate to pull down, followed by a maven coordinate, e.g. io.druid.extensions:mysql-metadata-storage
+
+`-h` or `--hadoop-coordinate`
+
+    Hadoop dependency to pull down, followed by a maven coordinate, e.g. org.apache.hadoop:hadoop-client:2.4.0
+
+`--no-default-hadoop`
+
+    Don't pull down the default hadoop coordinate, i.e., org.apache.hadoop:hadoop-client:2.3.0. If `-h` option is supplied, then default hadoop coordinate will not be downloaded.
+
+`-clean`
+    
+    Remove exisiting extension and hadoop dependencies directories before pulling down dependencies.
+
+`-l` or `--localRepository`
+
+    A local repostiry that Maven will use to put downloaded files. Then pull-deps will lay these files out into the extensions directory as needed.
+
+`-r` or `--remoteRepositories`
+
+    A JSON Array list of remote repositories to load dependencies from.
+
+`-d` or `--defaultVersion`
+
+    Version to use for extension artifacts without version information.
+
+To run `pull-deps`, you should
+
+1) Specify `druid.extensions.directory` and `druid.extensions.hadoopDependenciesDir`, these two properties tell `pull-deps` where to put extensions. If you don't specify them,  default values will be used, see [Configuration](../configuration/index.html).
+
+2) Tell `pull-deps` what to download using `-c` or `-h` option, which are followed by a maven coordinate.
+
+Example:
+
+Suppose you want to download ```druid-examples```, ```mysql-metadata-storage``` and ```hadoop-client```(both 2.3.0 and 2.4.0) with a specific version, you can run `pull-deps` command with `-c io.druid.extensions:druid-examples:0.8.0-rc1`, `-c io.druid.extensions:mysql-metadata-storage:0.8.0-rc1`, `-h org.apache.hadoop:hadoop-client:2.3.0` and `-h org.apache.hadoop:hadoop-client:2.4.0`, an example command would be:
+
+```java -classpath "/my/druid/library/*" io.druid.cli.Main tools pull-deps -clean -c io.druid.extensions:mysql-metadata-storage:0.8.0-rc1 -c io.druid.extensions:druid-examples:0.8.0-rc1 -h org.apache.hadoop:hadoop-client:2.3.0 -h org.apache.hadoop:hadoop-client:2.4.0```
```

Comment:
default maven version is optional for extension coordinate. I added more docs on this.